File: problems/python/arrays_hashing/longestConsecutiveSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        set1 = set()
        res = 0
        for num in nums:
            set1.add(num)
        for num in nums:
            if not (num-1 in set1):
                firstNum = num
                num = num+1
                while num in set1:
                    set1.remove(num)
                    num += 1
                res = max(res, num-firstNum)
        return res


        

File: java/arrays_hashing/longest_consecutive_sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
    	int res = 0;
    	if (nums == null) return 0;

    	for (int num: nums){
    		set.add(num);
        }

        for (int num: nums){
            int firstNum = num;
        	if(!set.contains(num-1)){
                while(set.contains(num)){
                    set.remove(num);
                	num++;
                }
                res = Math.max(res,(num-firstNum));
            }
        }

    	return res;
    }
}
